src/pages/data.test.js

```diff
@@ -0,0 +1,427 @@
+        })
+
+        screen.getByTestId('dhis2-ui-selectorbaritem').click()
+        const menuItems = await screen.findAllByTestId('dhis2-uicore-menuitem')
+        expect(menuItems).toHaveLength(2)
+        menuItems.map((menuItem, i) => {
+            expect(menuItem).toHaveTextContent(exchanges[i].displayName)
+        })
+    })
+
+    it('should display a genera message when no exchange is selected ', async () => {
+        const { screen } = setUp(<DataPage />)
+
+        expect(screen.getByTestId('entry-screen-message')).toBeInTheDocument()
+    })
+
+    it('should select and clear the selected exchange', async () => {
+        const anExchange = testDataExchange()
+        const exchanges = [anExchange, testDataExchange()]
+        const { screen } = setUp(<DataPage />, {
+            aggregateDataExchanges: exchanges,
+        })
+
+        screen.getByTestId('dhis2-ui-selectorbaritem').click()
+
+        const menuItems = await screen.findAllByTestId('dhis2-uicore-menuitem')
+        within(menuItems[0]).queryByText(anExchange.displayName).click()
+        expect(screen.getByTestId('data-exchange-selector')).toHaveTextContent(
+            anExchange.displayName
+        )
+
+        const headerBar = screen.getByTestId('dhis2-ui-selectorbar')
+        within(headerBar)
+            .queryByRole('button', { name: 'Clear selections' })
+            .click()
+        expect(screen.getByTestId('data-exchange-selector')).toHaveTextContent(
+            'Choose a data exchange'
+        )
+    })
+
+    it('should display the correct exchange specified in url if the param is present', async () => {
+        const anExchange = testDataExchange()
+        const exchanges = [anExchange, testDataExchange()]
+        const { screen } = setUp(<DataPage />, {
+            aggregateDataExchanges: exchanges,
+            exchangeId: anExchange.id,
+        })
+
+        expect(screen.getByTestId('data-exchange-selector')).toHaveTextContent(
+            anExchange.displayName
+        )
+    })
+
+    it('should display a preview table once an exchange id selected', async () => {
+        const anExchange = testDataExchange()
+
+        const exchanges = [anExchange]
+        const exchangeData = testDataExchangeSourceData()
+        const { screen } = setUp(<DataPage />, {
+            aggregateDataExchanges: exchanges,
+            exchangeId: anExchange.id,
+            exchangeData,
+        })
+
+        const titleBar = await screen.findByTestId('title-bar')
+        expect(titleBar).toHaveTextContent(anExchange.displayName)
+        expect(titleBar).toHaveTextContent(
+            `${anExchange.source.requests.length} data report`
+        )
+        const timeDifference = getRelativeTimeDifference({
+            startTimestamp: lastAnalyticsTableSuccess,
+            endTimestamp: serverDate,
+        })
+        expect(titleBar).toHaveTextContent(
+            `Source data was generated ${timeDifference} ago`
```

Comment:
maybe this would be good as a separate test in the sense that this `Source data was generated ${timeDifference} ago` time stamp will be the same for any exchange selected, but the rest of the test is more focused on making sure that the display matches the data for the specific exchange.

Reply:
I see your point, but i was thinking here it would be cleaner to keep the one user interaction all together, as the time differnece appears together  with the rest of the exchange data info . Otherwise  i think it becomes a bit hard to read?